File: src/BetterLyrics.Core/Interfaces/Services/IPlayHistoryService.cs
using BetterLyrics.Core.Models.Entities;
using BetterLyrics.Core.Models.Stats;

namespace BetterLyrics.Core.Interfaces.Services;

public interface IPlayHistoryService
{
    Task AddLogAsync(PlayHistoryItem item);
    Task<List<PlayHistoryItem>> GetRecentLogsAsync(int limit = 50);
    Task<List<PlayHistoryItem>> GetLogsByDateRangeAsync(DateTime start, DateTime end);

    Task<List<SongPlayCount>> GetTopSongsAsync(DateTime start, DateTime end, int limit = 10);
    Task<List<ArtistPlayCount>> GetTopArtistsAsync(DateTime start, DateTime end, int limit = 10);
    Task<TimeSpan> GetTotalListeningDurationAsync(DateTime start, DateTime end);
    Task<List<PlayerStats>> GetPlayerDistributionAsync(DateTime start, DateTime end);

    Task DeleteLogAsync(int id);
    Task ClearHistoryAsync();
    Task GenerateTestDataAsync(int count = 100);
}

File: src/BetterLyrics.KMP/shared/src/commonMain/kotlin/com/jayfunc/betterlyrics/interfaces/services/IPlayHistoryService.kt
package com.jayfunc.betterlyrics.interfaces.services

interface IPlayHistoryService {
    suspend fun initAsync()
    suspend fun getLogsByDateRangeAsync(start: Any, end: Any): List<Any>
    suspend fun addPlayLogAsync(id: String, playerId: String, title: String, artist: String, album: String, startedAt: Any, totalDurationMs: Long, durationPlayedMs: Long)
    suspend fun getTopSongsAsync(start: Any, end: Any, limit: Int = 10): List<Any>
    suspend fun getTopArtistsAsync(start: Any, end: Any, limit: Int = 10): List<Any>
    suspend fun getTotalListeningDurationAsync(start: Any, end: Any): Any
    suspend fun getPlayerDistributionAsync(start: Any, end: Any): List<Any>
    suspend fun incrementPlayerPlayCountAsync(playerId: String, count: Int = 1)
}
